From 3a3fdfbab108c649f93ceb396a3af3bfa264f1c6 Mon Sep 17 00:00:00 2001
From: Vignesh Lal T D <vdesikachary@maxlinear.com>
Date: Mon, 11 Nov 2024 19:30:27 +0530
Subject: [PATCH] WLANRTSYS-85958 Always update PMK cache for sibling link

Issue: After forget network on client, it sends authentication with SAE
for next connection request, the derived pmk is stored in assoc link pmk cache
but does not update on sibling link. Since an entry exists with same non-ap
mld mac address.

Fix: Always update pmk cache for sibling link.

FIXUP to WLANRTSYS-62845 MLO changes: MLO Auth Assoc, WPA3 and eapol changes
---
 src/ap/mxl_wpa_auth.c | 9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/src/ap/mxl_wpa_auth.c b/src/ap/mxl_wpa_auth.c
index 05130c04e..4a2ce0e77 100644
--- a/src/ap/mxl_wpa_auth.c
+++ b/src/ap/mxl_wpa_auth.c
@@ -755,14 +755,7 @@ void mxl_wpa_auth_pmksa_add_sae_sibling(struct wpa_authenticator *wpa_auth, cons
 			   const u8 *pmk, size_t pmk_len, const u8 *pmkid,
 			   int akmp)
 {
-	struct rsn_pmksa_cache_entry *pos;
-
-	pos = pmksa_cache_auth_get(wpa_auth->pmksa, addr, NULL);
-	if (!pos) {
-		wpa_auth_pmksa_add_sae(wpa_auth, addr, pmk, pmk_len, pmkid, akmp);
-	} else {
-		wpa_printf(MSG_DEBUG, "RSN: linked_sta PMK already cached");
-	}
+	wpa_auth_pmksa_add_sae(wpa_auth, addr, pmk, pmk_len, pmkid, akmp);
 }
 #endif /* CONFIG_IEEE80211BE_MXL_MLO */
 
-- 
2.43.0

